code/menuui/playermenu.cpp

```diff
@@ -1370,33 +1370,25 @@ DCF(bastion, "Temporarily sets the player to be on the Bastion (or any other mai
 	}
 }
 
-#define MAX_PLAYER_TIPS			40
-
-SCP_string Player_tips[MAX_PLAYER_TIPS];
-int Num_player_tips;
+SCP_vector<SCP_string> Player_tips;
 int Player_tips_shown = 0;
 
 // tooltips
-void player_tips_init()
+void parse_tips_table(const char* filename)
 {
-	Num_player_tips = 0;
 	
 	try
 	{
-		read_file_text("tips.tbl", CF_TYPE_TABLES);
+		read_file_text(filename, CF_TYPE_TABLES);
 		reset_parse();
 
 		while (!optional_string("#end")) {
 			required_string("+Tip:");
 
-			if (Num_player_tips >= MAX_PLAYER_TIPS) {
-				break;
-			}
-
 			SCP_string buf;
 			stuff_string(buf, F_MESSAGE);
-
-			Player_tips[Num_player_tips++] = message_translate_tokens(buf.c_str());
+			
+			Player_tips.push_back(message_translate_tokens(buf.c_str()));
```

Comment:
You shouldn't need to use a c_str() here because the vector already holds SCP_string.

Reply:
That's for `message_translate_tokens` which expects a char*

Reply:
That's what I get for reading too quick.  Nifty that it returns an SCP string.